discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -849,15 +861,19 @@ def process_user_bank_txs() -> None:
             # Note: while it's possible (even likely) to have multiple tx in the same slot,
             # these transactions can't be dependent on one another, so we don't care which order
             # we process them.
-            tx_infos.sort(key=lambda info: info[0]["result"]["slot"])
+            tx_infos.sort(key=lambda info: info[0].value.slot if info[0].value else 0)
 
             for tx_info, tx_sig in tx_infos:
                 if tx_info and last_tx_sig and last_tx_sig == tx_sig:
-                    last_tx = tx_info["result"]
+                    last_tx = tx_info.value
                 num_txs_processed += 1
 
-                tx_slot = tx_info["result"]["slot"]
-                timestamp = tx_info["result"]["blockTime"]
+                tx_value = tx_info.value
+                if tx_value is None:
+                    break
```

Comment:
this seems like a logic change - why do we stop processing if we encounter `None`?

Reply:
raising instead, that makes this the same logic. Great eye on this... Technically this will never happen to us, but in the case it did, raising will break us out of the flow, as it would have before. Notably though... this is different behavior than the other indexers which skip transactions. 

We should revisit this methodology overall. I know you had wanted to do some big refactors of all this code... and I agree!